src/core/UIHandler.ts

```diff
@@ -20,41 +20,30 @@ export class UIHandler extends EventTarget {
 	}
 
 	public bindUI(element: Element): void {
-		const selectors = [
-			`a${this.selector}`,
-			`input[type="submit"]${this.selector}`,
-			`input[type="image"]${this.selector}`,
-			`button[type="submit"]${this.selector}`,
-			`button[form]:not([type])${this.selector}`,
-			`form button:not([type])${this.selector}`,
-			`form${this.selector} input[type="submit"]`,
-			`form${this.selector} input[type="image"]`,
-			`form${this.selector} button[type="submit"]`,
-			`form${this.selector} button:not([type])`,
-		].join(', ');
-
-		const bindElement = (element: Element) => {
+		const selector = `a${this.selector}`;
+
+		const bindElement = (element: HTMLAnchorElement) => {
 			element.removeEventListener('click', this.handler);
 			element.addEventListener('click', this.handler);
 		};
 
-		const elements = element.querySelectorAll(selectors);
-		elements.forEach((element) => bindElement(element));
+		const elements = element.querySelectorAll(selector);
+		elements.forEach((element) => bindElement(element as HTMLAnchorElement));
 
-		if (element.matches(selectors)) {
-			bindElement(element);
+		if (element.matches(selector)) {
+			bindElement(element as HTMLAnchorElement);
 		}
```

Comment:
BTW: This looks suspicious to me. Shouldn't it be more like this?

```typescript
if (element.matches(selectors)) {
	bindElement(element);
} else {
	const elements = element.querySelectorAll(selectors);
	elements.forEach((element) => bindElement(element));
}
```
(The same applies to form binding below.)

Reply:
Done.